src/main/java/reactor/netty/channel/MonoSendMany.java

```diff
@@ -201,7 +204,7 @@ public void onSubscribe(Subscription s) {
 					@SuppressWarnings("unchecked") QueueSubscription<I> f =
 							(QueueSubscription<I>) s;
 
-					int m = f.requestFusion(Fuseable.ANY/* | Fuseable.THREAD_BARRIER*/);
+					int m = f.requestFusion(Fuseable.ANY | Fuseable.THREAD_BARRIER);
```

Comment:
Thread barrier should be here since if we have `publishOn` on the upstream, this will break fusion (we are rescheduling on the `eventLoop` and that might lead to incorrect fusion model and subsequent leaks since two threads can clean queue again)

Reply:
I commented that one specifically as it reduces a lot of the fusion scope unfortunately and I couldn't see at the time the impact. But that seems to be on a the safer side yes.